jsonpickle/unpickler.py

```diff
@@ -917,7 +929,9 @@ def _restore_dict(self, obj):
                 else:
                     str_k = k
                 self._namestack.append(str_k)
-                data[k] = self._restore(v)
+                data[k] = result = self._restore(v)
+                if isinstance(result, _Proxy):
```

Comment:
Is this fixing something found in the fuzzing or was the lack of it just causing a bug of some other sort?

Reply:
Yup, this edge case was found while fixing the  fuzzing errors. We end up leaving `_IDProxy` objects in the final output without this extra check.

This probably isn't too great from a performance perspective but we'd probably need some deeper restructuring to handle all of the edge cases in a more performant manner. e.g. either never return _IDProxy, or change the `def _restore(...)` return value to return `(object, bool is_proxy)` tuples to indicate that a proxy is being returned.